src/main/java/se/liu/ida/hefquin/engine/queryplan/utils/LogicalOpUtils.java

```diff
@@ -36,34 +44,31 @@
 import se.liu.ida.hefquin.engine.queryplan.physical.PhysicalOperator;
 import se.liu.ida.hefquin.engine.queryplan.physical.PhysicalOperatorForLogicalOperator;
 
-import java.util.ArrayList;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Set;
-
 public class LogicalOpUtils
 {
 	/**
 	 * Rewrites the given request operator (with a triple pattern request) into
 	 * a logical plan that uses the local vocabulary of the federation member of
 	 * the request.
 	 */
-	public static LogicalPlan rewriteToUseLocalVocabulary( final LogicalOpRequest<TriplePatternRequest, ?> reqOp ) {
-		final TriplePatternRequest tpReq = reqOp.getRequest();
-		final TriplePattern tp = tpReq.getQueryPattern();
-
+	public static LogicalPlan rewriteToUseLocalVocabulary( final LogicalOpRequest<?, ?> reqOp ) {
 		final FederationMember fm = reqOp.getFederationMember();
-		final VocabularyMapping vm = fm.getVocabularyMapping();
+		if (fm.getVocabularyMapping() == null) { // If no vocabulary mapping, nothing to translate.
+			return new LogicalPlanWithNullaryRootImpl(reqOp);
```

Comment:
I would prefer that we are not creating new `LogicalPlan` objects in such cases. In order to be able to avoid doing so, the argument to this function here cannot be a `LogicalOpRequest` but must be a `LogicalPlan` (or a `LogicalPlanWithNullaryRoot` if you want) that is assumed to have a `LogicalOpRequest` as its root operator.

Reply:
After changing the signature of this function in this way, it would perhaps be more logical to move the function directly into `ApplyVocabularyMappings`.